src/sdkV2.ts

```diff
@@ -0,0 +1,111 @@
+import { ethers } from 'ethers';
+import { EvmAddress, TxOverrides } from './types';
+import { GasStation } from './GasStation';
+
+import Erc20Abi from './abi/ERC20.json';
+import IDCSEntryAbi from './abiV2/IDCSEntry.json';
+
+export default class CegaEvmSDKV2 {
+  private _provider: ethers.providers.Provider;
+
+  private _signer: ethers.Signer | undefined;
+
+  private _gasStation: GasStation;
+
+  private _cegaEntryAddress: EvmAddress;
+
+  constructor(
+    cegaEntryAddress: EvmAddress,
+    gasStation: GasStation,
+    provider: ethers.providers.Provider,
+    signer: ethers.Signer | undefined = undefined,
+  ) {
+    this._provider = provider;
+    this._signer = signer;
+    this._gasStation = gasStation;
+    this._cegaEntryAddress = cegaEntryAddress;
+  }
+
+  setProvider(provider: ethers.providers.Provider) {
+    this._provider = provider;
+  }
+
+  setSigner(signer: ethers.Signer) {
+    this._signer = signer;
+  }
+
+  setCegaEntryAddress(cegaEntryAddress: EvmAddress) {
+    this._cegaEntryAddress = cegaEntryAddress;
+  }
+
+  loadCegaEntry(): ethers.Contract {
+    return new ethers.Contract(
+      this._cegaEntryAddress,
+      IDCSEntryAbi.abi,
+      this._signer || this._provider,
+    );
+  }
+
+  async getProductDcs(productId: ethers.BigNumberish) {
```

Comment:
Which one is this product ID? Just trying to make sure cuz there's a few variants (ie supabase config one)

Reply:
it would be the supabase `contract_product_uid` for v2

but the value in supabase will just be a stringified version of an integer, not a normal string descriptor like it is for v1 products

Reply:
can we add a JSDoc description here so that we can see in the IDE what type of ID we need to send here? 🙏🏼

Reply:
discussed offline with @consvic . will add the JSDoc descriptions later. doesn't seem needed here

for clarity, with the SDK, we will use product IDs as denoted by the SC, not internal/supabase product IDs. this is because other people that use our SDK won't have knowledge of our supabase ids